root/report/components/ArtistList.js

```diff
@@ -10,51 +10,47 @@
 import * as React from 'react';
 
 import PaginatedResults from '../../components/PaginatedResults';
-import EntityLink from '../../static/scripts/common/components/EntityLink';
-import loopParity from '../../utility/loopParity';
+import Table from '../../components/Table';
+import {
+  defineNameColumn,
+  defineTypeColumn,
+} from '../../utility/tableColumns';
 import type {ReportArtistT} from '../types';
 
 type Props = {
   +items: $ReadOnlyArray<ReportArtistT>,
   +pager: PagerT,
 };
 
-const ArtistList = ({
+const ArtistReportList = ({
   items,
   pager,
-}: Props): React.Element<typeof PaginatedResults> => (
-  <PaginatedResults pager={pager}>
-    <table className="tbl">
-      <thead>
-        <tr>
-          <th>{l('Artist')}</th>
-          <th>{l('Type')}</th>
-        </tr>
-      </thead>
-      <tbody>
-        {items.map((item, index) => (
-          <tr className={loopParity(index)} key={item.artist_id}>
-            {item.artist ? (
-              <>
-                <td>
-                  <EntityLink entity={item.artist} />
-                </td>
-                <td>
-                  {nonEmpty(item.artist.typeName)
-                    ? lp_attributes(item.artist.typeName, 'artist_type')
-                    : l('Unknown')}
-                </td>
-              </>
-            ) : (
-              <td colSpan="2">
-                {l('This artist no longer exists.')}
-              </td>
-            )}
-          </tr>
-        ))}
-      </tbody>
-    </table>
-  </PaginatedResults>
-);
+}: Props): React.Element<typeof PaginatedResults> => {
+  const existingArtists = items
+    .filter(item => item.artist != null)
+    .map(item => item.artist);
+  const columns = React.useMemo(
+    () => {
+      const nameColumn = defineNameColumn<ArtistT>({
+        title: l('Artist'),
+      });
+      const typeColumn = defineTypeColumn({
+        typeContext: 'artist_type',
+      });
```

Comment:
Not sure whether it would be better to use the simplified approach I took here (just map the artists first) or just use the same column definitions as in the ArtistAnnotationList.

Reply:
If it can save lines of code and make maintenance easier, sharing the column definitions seems like a good idea. (But that may not be true until more are converted -- just looking at these two I'd say either approach is fine.)